### PR TITLE
Minor cleanups

### PR DESCRIPTION
**Pull request by fastier-li**
Minor cleanups, usually suggested by IDEA/Scala.

### CODE REVIEW

File: photon-ml/src/main/scala/com/linkedin/photon/ml/util/Logging.scala [hunk 1]
```diff
@@ -25,13 +25,13 @@ import org.slf4j.{Logger, LoggerFactory}
  * https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/internal/Logging.scala
  */
 protected[ml] trait Logging {
-  // Transient so that it can be used distributed code without incurring unecessary serialization / io
-  @transient private var log: Logger = null
+  // Transient so that it can be used distributed code without incurring unnecessary serialization / io
+  @transient private var log: Logger = _
```
Comment (joshvfleming) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/util/Logging.scala]: Let's leave this as "null". I know it's not idiomatic, but I kept it this way on purpose to more clearly match the condition on L43.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/stat/BasicStatistics.scala [hunk 2]
```diff
@@ -30,6 +30,7 @@ private[ml] object BasicStatistics {
   /**
    * Generate basic statistics (e.g., max, min, mean, variance) of [[data.LabeledPoint]] RDD using the mllib
    * interface.
+    *
```
Comment (joshvfleming) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/stat/BasicStatistics.scala]: Extra space.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/util/Logging.scala (hunk 1 above)
Comment (fastier-li) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/util/Logging.scala]: OK.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/stat/BasicStatistics.scala (hunk 2 above)
Comment (fastier-li) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/stat/BasicStatistics.scala]: I'm on it.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/training/Driver.scala [hunk 3]
```diff
@@ -71,6 +69,8 @@ final class Driver(val params: GameParams, val sparkContext: SparkContext, val l
       data
     }
 
+    logger.info(s"Time elapsed after computing features summary: ${timer.durationSeconds} (s)\n")
```
Comment (joshvfleming) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/training/Driver.scala]: This seems out of place in this version of the code. Did you mean to commit this line?

Comment (fastier-li) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/training/Driver.scala]: Ah yes! Taking it out.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/util/Logging.scala [hunk 4]
```diff
@@ -24,14 +24,16 @@ import org.slf4j.{Logger, LoggerFactory}
  *
  * https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/internal/Logging.scala
  */
+//noinspection ConvertNullInitializerToUnderscore
```
Comment (ashelkovnykov) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/util/Logging.scala]: Is this leftover from something?

**Pull request merged**